### PR TITLE
PAAS-4603 expire pull requests cache so updated risks are visible

### PR DESCRIPTION
**Pull request by grosser**
found this while poking at risks vs commit status ticket

@zendesk/compute

### CODE REVIEW

File: test/models/changeset/pull_request_test.rb [hunk 1]
```diff
@@ -70,15 +79,17 @@ def no_risks
       {
         "number" => 1,
         "pull_request" => {
-          "state" => 'open',
-          "body" => 'pr description [samson review]'
+          "state" => "open",
+          "body" => "pr description [samson review]"
         },
-        "github" => {
-          "action" => 'opened'
```
Comment (grosser) [inline test/models/changeset/pull_request_test.rb]: this test was just wrong, action is directly in the payload 🤷‍♂

File: app/models/changeset.rb [hunk 2]
```diff
@@ -95,6 +95,7 @@ def compare_cache_key
 
   # github only supports finding open PRs for branches (not commits or tags)
   # https://help.github.com/en/articles/searching-issues-and-pull-requests
+  # TODO: can we just use .new here to avoid extra fetch ?
```
Comment (adammw) [inline app/models/changeset.rb]: relevant?

File: app/models/changeset/commit.rb [hunk 3]
```diff
@@ -36,6 +36,7 @@ def short_sha
     @data.sha.slice(0, 7)
   end
 
+  # @return [Integer, NilClass]
```
Comment (adammw) [inline app/models/changeset/commit.rb]:
```suggestion
  # @return [Integer|NilClass]
```

File: app/models/changeset/pull_request.rb [hunk 4]
```diff
@@ -20,43 +20,54 @@ class Changeset::PullRequest
...
-      previous_desc = payload.dig('changes', 'body', 'from')
-      return false if !previous_desc || (previous_desc =~ WEBHOOK_FILTER && data['body'] =~ WEBHOOK_FILTER)
+      data['body'].match? WEBHOOK_FILTER
     end
 
-    !!(data['body'] =~ WEBHOOK_FILTER)
+    private
+
+    def cache_key(repo, number)
+      [self, repo, number].join("-")
```
Comment (adammw) [inline app/models/changeset/pull_request.rb]: does repo include org name?

File: test/models/changeset/pull_request_test.rb [hunk 5]
```diff
@@ -44,6 +44,15 @@ def no_risks
     end
   end
 
+  describe ".expire" do
+    it "expires find cache" do
+      GITHUB.expects(:pull_request).with("foo/bar", 42).returns({}).times(2)
+      2.times { Changeset::PullRequest.find("foo/bar", 42) }
+      Changeset::PullRequest.expire("foo/bar", 42)
+      Changeset::PullRequest.find("foo/bar", 42)
```
Comment (adammw) [inline test/models/changeset/pull_request_test.rb]: assert return value?

File: app/models/changeset/pull_request.rb (hunk 4 above)
Comment (grosser) [reply app/models/changeset/pull_request.rb]: yep

File: app/models/changeset/commit.rb (hunk 3 above)
Comment (grosser) [reply app/models/changeset/commit.rb]: only guidlines I found say to use comma `# @param argname [true, false] only true or false
` https://rubydoc.info/gems/yard/file/docs/GettingStarted.md

File: app/models/changeset.rb (hunk 2 above)
Comment (grosser) [reply app/models/changeset.rb]: that's for a later PR :)

File: test/models/changeset/pull_request_test.rb (hunk 5 above)
Comment (grosser) [reply test/models/changeset/pull_request_test.rb]: done

**Pull request merged**